lib/Optimizer/CodeGen/LowerToQIRProfile.cpp

```diff
@@ -380,57 +395,70 @@ struct CallAlloc : public OpRewritePattern<LLVM::CallOp> {
+  QIRToQIRProfileQIRPass(llvm::StringRef convertTo_) : QIRToQIRProfileBase() {
+    convertTo.setValue(convertTo_);
+  }
 
   void runOnOperation() override {
     auto *op = getOperation();
-    LLVM_DEBUG(llvm::dbgs() << "Before base profile:\n" << *op << '\n');
+    LLVM_DEBUG(llvm::dbgs() << "Before QIR profile:\n" << *op << '\n');
     auto *context = &getContext();
     RewritePatternSet patterns(context);
-    // Note: LoadMeasureResult is not compliant with the Base Profile
+    // Note: LoadMeasureResult is not compliant with the Base Profile, so don't
+    // add it here unless we're specifically doing the Adaptive Profile.
     patterns.insert<AddrOfCisToBase, ArrayGetElementPtrConv, CallAlloc,
                     CalleeConv, EraseArrayAlloc, EraseArrayRelease,
                     EraseDeadArrayGEP, MeasureCallConv,
                     MeasureToRegisterCallConv, XCtrlOneTargetToCNot>(context);
     if (failed(applyPatternsAndFoldGreedily(op, std::move(patterns))))
       signalPassFailure();
-    LLVM_DEBUG(llvm::dbgs() << "After base profile:\n" << *op << '\n');
+    LLVM_DEBUG(llvm::dbgs() << "After QIR profile:\n" << *op << '\n');
   }
 
 private:
   FrozenRewritePatternSet patterns;
 };
 } // namespace
 
-std::unique_ptr<Pass> cudaq::opt::createQIRToBaseProfilePass() {
-  return std::make_unique<QIRToBaseProfileQIRPass>();
+std::unique_ptr<Pass>
+cudaq::opt::createQIRToQIRProfilePass(llvm::StringRef convertTo) {
+  return std::make_unique<QIRToQIRProfileQIRPass>(convertTo);
 }
 
 //===----------------------------------------------------------------------===//
 
 namespace {
-/// Base Profile Preparation:
+/// QIR Profile Preparation:
 ///
-/// Before we can do the conversion to the QIR base profile with different
-/// threads running on different functions, the module is updated with the
-/// signatures of functions from the QIR ABI that may be called by the
-/// translation. This trivial pass only does this preparation work. It performs
-/// no analysis and does not rewrite function body's, etc.
+/// Before we can do the conversion to the QIR profile with different threads
+/// running on different functions, the module is updated with the signatures of
+/// functions from the QIR ABI that may be called by the translation. This
+/// trivial pass only does this preparation work. It performs no analysis and
+/// does not rewrite function body's, etc.
 
 static const std::vector<std::string> measurementFunctionNames{
     cudaq::opt::QIRMeasureBody, cudaq::opt::QIRMeasure,
     cudaq::opt::QIRMeasureToRegister};
 
-struct BaseProfilePreparationPass
-    : public cudaq::opt::QIRToBaseQIRPrepBase<BaseProfilePreparationPass> {
+struct QIRProfilePreparationPass
+    : public cudaq::opt::QIRToQIRProfilePrepBase<QIRProfilePreparationPass> {
+
+  explicit QIRProfilePreparationPass(llvm::StringRef convertTo_)
+      : QIRToQIRProfilePrepBase() {
+    convertTo.setValue(convertTo_);
```

Comment:
What is convertTo used for here?

Reply:
It's not currently used for anything in this particular PR, but it _will_ be used to conditionally call `createLLVMFunctionSymbol()` in `runOperation()` for the `cudaq::opt::QIRReadResultBody` symbol.

Reply:
OK. But, I don't think adding any function symbol as a declaration needs to be conditional. Unused functions should get pruned out in LLVM-IR dialect to LLVM IR proper step.

Reply:
OK, I will remove this one.